lib/vanagon/patch.rb

```diff
@@ -18,21 +31,28 @@ class Patch
     #   @return [String] What step should this patch be applied to, one of ["unpack", "install"]
     attr_reader :after
 
-    # @!attribute [r] destination
-    #   @return [String] The working directory where this patch will be applied. Only used for post-installation patches.
-    attr_reader :destination
+    def initialize(origin_path, options = {}) # rubocop:disable Metrics/AbcSize
+      valid_keys = %i[namespace destination strip fuzz after]
+      bad_keys = options.each_key.reject { |k| valid_keys.include? k }
+
+      unless bad_keys.empty?
+        raise Vanagon::Error, "Bad options in patch initialization: #{bad_keys}."
+      end
+      unless ['unpack', 'install'].include?(options[:after])
+        raise Vanagon::Error, 'Only "unpack" or "install" permitted for "after" option.'
+      end
 
-    def initialize(path, strip, fuzz, after, destination)
-      raise Vanagon::Error, "We can only apply patches after the source is unpacked or after installation" unless ['unpack', 'install'].include?(after)
-      @path = path
-      @strip = strip
-      @fuzz = fuzz
-      @after = after
-      @destination = destination
+      @origin_path = origin_path
+      @namespace = options[:namespace]
+      @assembly_path = "patches/#{@namespace}/#{File.basename(@origin_path)}"
+      @strip = options[:strip]
+      @fuzz = options[:fuzz]
+      @after = options[:after]
+      @destination = options[:destination]
```

Comment:
Would it be better for defaults to get set here vs. in the `apply_patch` method? In my head, I think it makes more sense to set these on initialization, but I'm not sure if making that change would have other consequences.

Reply:
Tried it that way, passes rspec. I like it.